src/doppler-security/JwtVerifier.ts

```diff
@@ -0,0 +1,47 @@
+import { JsonWebTokenError, JwtPayload } from "jsonwebtoken";
+import { Result } from "src/shared/result-types";
+import { TokenData } from "./TokenData";
+import { verify } from "jsonwebtoken";
+
+export class JwtVerifier {
+  private readonly _publicKey: string;
+
+  constructor({ publicKey }: { publicKey: string }) {
+    this._publicKey = publicKey;
+  }
+
+  public verify(token: string): Result<TokenData, JsonWebTokenError> {
+    try {
+      const payload = verify(token, this._publicKey, { complete: false });
+
+      if (typeof payload !== "string" && !payload["exp"]) {
```

Comment:
No sería mas claro:
```suggestion
      if (payload === null || typeof payload !== "object" || !payload["exp"]) {
```
(seguramente no, y debe haber un por que acerca de la decisión, pero me confunde un poco al leer en la condicion el `typeof payload !== "string"` 😄)

Reply:
The condition is because of the possible result types:

<img width="1102" alt="image" src="https://user-images.githubusercontent.com/1157864/198012968-424662ad-a9cc-45c9-8919-7b7704ec919c.png">

But you are right, it is weird and I think that I can do it better.

Reply:
Done! could you review it again?

Reply:
The same changes are here: https://github.com/FromDoppler/doppler-system-usage/pull/4